src/main/java/com/iexec/common/docker/DockerRunRequest.java

```diff
@@ -72,30 +73,37 @@ public String getStringArgsCmd() {
         return devices != null ? new ArrayList<>(devices) : List.of();
     }
 
-    // override builder's isSgx() & devices() methods
+    public SgxDriverMode getSgxDriverMode() {
+        return sgxDriverMode != null ? sgxDriverMode : SgxDriverMode.NONE;
+    }
+
+    // override builder's sgxDriverMode() & devices() methods
     public static class DockerRunRequestBuilder { 
-        private boolean isSgx;
+        private SgxDriverMode sgxDriverMode;
         private List<Device> devices;
 
         /**
          * Add an SGX device when isSgx is true.
          * 
-         * @param isSgx
+         * @param sgxDriverMode SGX driver mode
```

Comment:
Could you describe in the javadoc comment what happens depending on the value of `sgxDriverMode` ?

Reply:
Sure. JavaDoc is currently wrong as it still refers to `isSgx` which has been removed. Let's update it!

Reply:
Done in https://github.com/iExecBlockchainComputing/iexec-common/pull/308/commits/24b0c04479f5ce6a5963a0c09882a81f478e7c41.